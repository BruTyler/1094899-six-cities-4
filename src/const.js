export const PlaceType = {
  APARTMENT: `Apartment`,
  PRIVATE_ROOM: `Private room`,
};
