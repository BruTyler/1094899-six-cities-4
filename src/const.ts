export enum PlaceType {
  APARTMENT = `Apartment`,
  ROOM = `Private room`,
  HOUSE = `House`,
  HOTEL = `Hotel`,
};

export enum SortType {
  POPULAR = `Popular`,
  PRICE_LOW = `Price: low to high`,
  PRICE_HIGH = `Price: high to low`,
  TOP_RATED = `Top rated first`,
};

export enum AuthorizationStatus {
  NO_AUTH = `NO_AUTH`,
  AUTH = `AUTH`,
};

export enum BuisnessRequirements {
  MAX_PHOTOS_PER_APARTMENT = 6,
  MAX_REVIEWS_PER_APARTMENT = 10,
  MIN_REVIEW_TEXT_LENGTH = 50,
  MAX_REVIEW_TEXT_LENGTH = 300,
};

export enum AppRoute {
  AUTH = `/login`,
  ROOT = `/`,
  FAVORITES = `/favorites`,
  PROPERTY_WITH_ID = `/offer/:id`,
};

export enum ApartmentEnvironment {
  FAVORITE = `favorites`,
  NEARBY_PLACES = `near-places`,
  MAIN_WINDOW = `cities`
};

export enum MapEnvironment {
  NEARBY_PLACES = `property__map`,
  MAIN_WINDOW = `cities__map`
};
