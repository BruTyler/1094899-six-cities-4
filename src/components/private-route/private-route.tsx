import * as React from 'react';
import {Route, Redirect, RouteProps} from 'react-router-dom';
import {connect} from 'react-redux';
import {AppRoute, AuthorizationStatus} from '../../const';
import {getAuthorizationStatus} from '../../reducer/user/selectors';

type Props = RouteProps & {
  authStatus: AuthorizationStatus;
  render: () => React.ReactNode;
}

const PrivateRoute: React.FunctionComponent<Props> = ({render, path, exact, authStatus}: Props) => (
  <Route
    path={path}
    exact={exact}
    render={() => {
      return (
        authStatus === AuthorizationStatus.AUTH
          ? render()
          : <Redirect to={AppRoute.AUTH} />
      );
    }}
  />
);

const mapStateToProps = (state) => ({
  authStatus: getAuthorizationStatus(state),
});

export {PrivateRoute};
export default connect(mapStateToProps)(PrivateRoute);
