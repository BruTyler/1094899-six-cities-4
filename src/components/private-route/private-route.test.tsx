import React from 'react';
import {Route, MemoryRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';
import {PrivateRoute} from './private-route.jsx';
import {AuthorizationStatus, AppRoute} from '../../const.js';

const routeTravelling = (userAuthStatus) => (
  <MemoryRouter initialEntries={[`/secret`]} initialIndex={0}>
    <Route exact
      path={AppRoute.AUTH}
      render={() => <div>login page</div>}
    />
    <PrivateRoute exact
      path="/secret"
      render={() => <div>private page</div>}
      authStatus={userAuthStatus}
    />
  </MemoryRouter>
);

describe(`<PrivateRoute /> render suit`, () => {
  it(`<PrivateRoute /> enables rendering private page for authenticated user`, () => {
    const generatedTree = renderer.create(
        routeTravelling(AuthorizationStatus.AUTH)
    ).toJSON();

    expect(generatedTree).toMatchSnapshot();
  });

  it(`<PrivateRoute /> redirects non-authenticated user to login page`, () => {
    const generatedTree = renderer.create(
        routeTravelling(AuthorizationStatus.NO_AUTH)
    ).toJSON();

    expect(generatedTree).toMatchSnapshot();
  });
});
