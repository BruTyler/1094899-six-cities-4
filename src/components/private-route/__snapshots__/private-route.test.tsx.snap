// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrivateRoute /> render suit <PrivateRoute /> enables rendering private page for authenticated user 1`] = `
<div>
  private page
</div>
`;

exports[`<PrivateRoute /> render suit <PrivateRoute /> redirects non-authenticated user to login page 1`] = `
<div>
  login page
</div>
`;
